testutils/test_utils.go

```
@@ -135,17 +135,24 @@ func VerifyRename(t *testing.T, source, target string) {
 func SetEnv(t *testing.T, envar, value string) func() {
 	t.Helper()
 
-	old := os.Getenv(envar)
+	old, reset := os.LookupEnv(envar)
 
 	err := os.Setenv(envar, value)
 	if err != nil {
 		t.Fatalf("setting %s environment variable to %s", envar, value)
 	}
 
 	return func() {
-		err := os.Setenv(envar, old)
-		if err != nil {
-			t.Fatalf("setting %s environment variable to %s", envar, old)
+		if reset {
+			err := os.Setenv(envar, old)
+			if err != nil {
+				t.Fatalf("setting %s environment variable to %s", envar, old)
+			}
+		} else {
+			err := os.Unsetenv(envar)
+			if err != nil {
+				t.Fatalf("unseeting %s environment variable", envar)
```

This PR didn't introduce the problem, but the actual error is lost in these `Fatalf` messages.

---

@jchampio what do you mean.  Note I just modified the error messages to allow distinguishing between all three error cases in this function.

---

I see...I need to print out the `err`.  Done.